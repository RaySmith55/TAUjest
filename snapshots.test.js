var itemStock = [
    {'id': '1', 'itemName': 'Razors', 'Stock': '10'},
    {'id': '2', 'itemName': 'Socks', 'Stock': '1'},
    {'id': '3', 'itemName': 'Towels', 'Stock': '100'},
    {'id': '4', 'itemName': 'Socks', 'Stock': '100'}
];

function filterItemsStock(arr, key, term) {
    return arr.filter(function(obj) {
        return obj[key] === term
    });}
    
test('it returns all items with matching id', () => {
    expect(filterItemsStock(itemStock, 'id', '1')).toEqual([
        {'id': '1', 'itemName': 'Razors', 'Stock': '10'}
    ])
});

test('it returns all items with matching item name', () => {
    expect(filterItemsStock(itemStock, 'itemName', 'Socks')).toEqual([
        {'id': '2', 'itemName': 'Socks', 'Stock': '1'},
        {'id': '4', 'itemName': 'Socks', 'Stock': '100'}
    ])
});
