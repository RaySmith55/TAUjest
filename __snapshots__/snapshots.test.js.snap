// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it returns all items with matching id 1`] = `
Array [
  Object {
    "Stock": "10",
    "id": "1",
    "itemName": "Razors",
  },
]
`;

exports[`it returns all items with matching item name 1`] = `
Array [
  Object {
    "Stock": "1",
    "id": "2",
    "itemName": "Socks",
  },
  Object {
    "Stock": "100",
    "id": "4",
    "itemName": "Socks",
  },
]
`;
